apps/labs/posts/pydata-theme-a11y-fixes.mdx

```diff
@@ -0,0 +1,479 @@
+
+If you're curious about how I generated the screenshots in the pull request that
+I linked, or if you're curious about how your own website looks in terms of
+landmarks, you can download a browser extension like
+[WAVE](https://wave.webaim.org/extension/) or [Accessibility
+Insights](https://accessibilityinsights.io/downloads/). These extensions provide
+a way to view all of the landmarks on a web page at once.
+
+## Sidebar Table of Contents, Before and After
+
+This section assumes some knowledge of HTML and CSS.
+
+A crucial part of web accessibility is keyboard accessibility. This means that
+users should be able to use your website without a mouse or touch device—in
+other words, with only a keyboard. Not only does this help low-vision users;
+providing full keyboard access is like providing a kind of input API to your
+site. Different kinds of assistive tech can hook into the same interface
+intended for keyboards. (There is a nice historical continuity here as some of
+the first [typewriters were assistive
+tech](https://en.wikipedia.org/wiki/Hansen_Writing_Ball).)
+
+The PyData theme provides a table of contents in the left sidebar. One nice
+feature of this table of contents is that site authors can choose how many
+nested levels to show before the sublevels are hidden automatically beneath an
+expand/collapse button. The only problem with this expand/collapse "button",
+though, was that it was not keyboard accessible.
+
+![A "before" screenshot shows a mouse pointer hovering over a toggle "button" to
+convey the idea that prior to the accessibility work, the "button" was only
+point-and-click accessible, not
+keyboard accessible](/posts/pydata-theme-a11y-fixes/sidebar-toc-before.png)
+
+I put "button" in quotes because a true HTML `<button>` is keyboard accessible,
+meaning that a user can both navigate to it using only the keyboard (via the
+`Tab` key) **and** a user can "press" the button using only the keyboard (via
+`Space`, `Enter` or `Return` keys). But in the case of our foldable/unfoldable
+table of contents, it had been implemented without HTML buttons to avoid
+JavaScript. It used two tricks in a clever way:
+
+1. In HTML, if you associate a `<label>` with an `<input type="checkbox" />`,
+   then clicking the label checks and unchecks the associated checkbox.
+2. In CSS, you can select a checked checkbox using the `:checked`
+   pseudo-selector.
+
+You can then use CSS to always hide the checkbox and to selectively fold or
+unfold the next level in the table of contents based on whether the checkbox is
+checked. In code, it looks like the following:
+
+```html
+<li>
+  <!-- table of contents entry --!>
+  <a href="kitchen-sink">Kitchen Sink</a>
+
+  <!-- checkbox hidden via CSS -->
+  <input id="toc-checkbox-1" class="toc-checkbox" type="checkbox" />
+
+  <!-- label associated with checkbox -->
+  <label for="toc-checkbox-1">🔽</label>
+
+  <!-- next level of TOC -->
+  <ul>
+    <!-- ... -->
+  </ul>
+</li>
+```
```

Comment:
Not for here, but I'm not sre we should have dark code block for a light theme blog.